ui/raidboss/data/triggers/o9s.js

```diff
@@ -182,16 +151,365 @@
+          return 5;
+        }
+        if ( (data.role != 'tank' && data.role != 'healer') && data.phaseType == 'fire' ) {
+          // 24s duration on entropy
+          if ( data.entropyCount > 1 )
+            return 12;
+          return 19;
+        }
+        if ( data.phaseType == 'enrage' ) {
+          // 6s for everyone on enrage
+          return 1;
+        }
+      },
+      alertText: function(data) {
+        if ( data.entropyCount == 1 && (data.role == 'tank' || data.role == 'healer') ) {
+          return {
+            en: 'Spread (Tanks/Healers)',
+            de: 'Verteilen (Tanks/Heiler)',
+            fr: 'Ecartez-vous (Tanks/Healers)',
+          };
+        }
+        if ( data.entropyCount == 2 && (data.role == 'tank' || data.role == 'healer' ) ) {
+          return {
+            en: 'Spread and stay!(Tanks/Healers)',
+            de: 'Verteilen und bleiben (Tanks/Heiler)',
+            fr: 'Ecartez-vous et rester (Tanks/Healers)',
+          };
+        }
+        if ( data.entropyCount == 1 && (data.role != 'tank' && data.role != 'healer') ) {
+          return {
+            en: 'Spread (DPS)',
+            de: 'Verteilen (DPS)',
+            fr: 'Ecartez-vous (DPS)',
+          };
+        }
+        if ( data.entropyCount == 2 && (data.role != 'tank' && data.role != 'healer') ) {
+          return {
+            en: 'Stack and Stay! (DPS)',
+            de: 'Stack und Bleiben! (DPS)',
+            fr: 'Empiler et rester! (DPS)',
+          };
+        }
+        if ( data.phaseType == 'enrage' ) {
+          return {
+            en: 'Spread (Everyone)',
+            de: 'Verteilen (Jeder)',
+            fr: 'Ecartez-vous (Tout le monde)',
+          };
+        }
+      },
+      run: function(data) {
+        if ( data.entropyCount > 1 )
+          delete data.entropyCount;
+        if ( data.phaseType == 'orb' || data.phaseType == 'enrage' )
+          delete data.entropyCount;
+      },
+    },
+    {
+      id: 'O9S entropy avoid hit',
+      regex: /:(\y{Name}) gains the effect of (?:Unknown_640|Entropy) from  for (\y{Float}) Seconds/,
+      regexDe: /:(\y{Name}) gains the effect of (?:Unknown_640|Chaosflammen) from  for (\y{Float}) Seconds/,
+      regexFr: /:(\y{Name}) gains the effect of (?:Unknown_640|Flammes du chaos) from  for (\y{Float}) Seconds/,
+      regexJa: /:(\y{Name}) gains the effect of (?:Unknown_640|混沌の炎) from  for (\y{Float}) Seconds/,
+      condition: function(data, matches) {
+        return matches[1] == data.me;
+      },
+      preRun: function(data) {
+        data.entropyCount = data.entropyCount || 0;
+      },
+      delaySeconds: function(data) {
+        if ( (data.role == 'tank' || data.role == 'healer') && data.phaseType == 'fire' )
+          return 19;
+        if ( (data.role != 'tank' && data.role != 'healer') && data.phaseType == 'fire' )
+          return 5;
+      },
+      infoText: function(data) {
+        if ( data.phaseType == 'fire' ) {
```

Comment:
Thanks for this too.  I was going to look at fixing the "Hide middle" during orb phase, but you've already fixed that!